elements/lisk-chain/test/unit/transactions.spec.ts

```diff
@@ -11,16 +11,66 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+import { getRandomBytes } from '@liskhq/lisk-cryptography';
+import { Transaction } from '../../src/transaction';
+
 describe('blocks/transactions', () => {
 	describe('transaction', () => {
 		it.todo('should have id');
 		it.todo('should have senderAddress');
 		it.todo('should throw when moduleID is below 2');
 		it.todo('should throw when moduleID is invalid');
-		it.todo('should throw when assetID is invalid');
+		it.todo('should throw when commandID is invalid');
 		it.todo('should throw when sender public key is invalid');
 		it.todo('should throw when nonce is invalid');
 		it.todo('should throw when fee is invalid');
-		it.todo('should throw when asset is invalid');
+		it.todo('should throw when params is invalid');
+	});
+	describe('#validateTransaction', () => {
+		let transaction: Transaction;
+
+		it('should throw when sender public key is not 32 bytes', () => {
+			transaction = new Transaction({
+				moduleID: 2,
+				commandID: 0,
+				fee: BigInt(613000),
+				params: getRandomBytes(500),
+				nonce: BigInt(2),
+				senderPublicKey: getRandomBytes(31),
+				signatures: [getRandomBytes(64)],
+			});
+			expect(() => transaction.validate()).toThrow('Lisk validator found 1 error[s]');
+		});
+
+		it('should throw when signatures is empty', () => {
+			transaction = new Transaction({
+				moduleID: 2,
+				commandID: 0,
+				fee: BigInt(613000),
+				params: getRandomBytes(500),
+				nonce: BigInt(2),
+				senderPublicKey: getRandomBytes(32),
+				signatures: [],
+			});
+			expect(() => transaction.validate()).toThrow('Signatures must not be empty');
+		});
+
+		it('should throw when any of signatures are not 64 bytes', () => {
+			transaction = new Transaction({
+				moduleID: 2,
+				commandID: 0,
+				fee: BigInt(613000),
+				params: getRandomBytes(500),
+				nonce: BigInt(2),
+				senderPublicKey: getRandomBytes(32),
+				signatures: [
+					getRandomBytes(64),
+					getRandomBytes(32),
+					getRandomBytes(32),
+					getRandomBytes(64),
+				],
+			});
+			expect(() => transaction.validate()).toThrow('Signature must be empty or 64 bytes');
```

Comment:
this test expect refers to `should throw when any of signatures are not 64 bytes`?

Reply:
ah, i got confused `signature`-`s`, maybe error message could be changed `Signature must always be of 64 bytes`
as empty signature is also not allowed i think, or is it?
```ts
signatures: [
					getRandomBytes(64),
					getRandomBytes(32),
					getRandomBytes(32),
					getRandomBytes(64),
					Buffer.from(''),
				],
```

Reply:
empty signature is allowed in case of multi-signature.
when signatures are `[mandatory, mandatory, optional, optional, optional]` and `numberOfSignatures == 4` one of optional must be empty